pkg/controllers/dashboard/helm/repo_oci.go

```diff
@@ -61,18 +77,41 @@ func (o *OCIRepohandler) onClusterRepoChange(key string, clusterRepo *catalog.Cl
 	if clusterRepo == nil {
 		return nil, nil
 	}
-
 	// Ignore non OCI ClusterRepos
 	if !registry.IsOCI(clusterRepo.Spec.URL) {
 		return clusterRepo, nil
 	}
 
+	// this is to prevent the handler from making calls when the crd is outdated.
+	updatedRepo, err := o.clusterRepoController.Get(key, metav1.GetOptions{})
+	if err == nil && updatedRepo.ResourceVersion != clusterRepo.ResourceVersion {
+		return clusterRepo, nil
+	}
+
+	if shouldResetRetries(clusterRepo) {
```

Comment:
I don't think so. If the max number of retries is 5, the 5th retry should still work. Also, we can't reset when reaching max retry without checking for other conditions otherwise it will retry forever